### ISSUE TITLE
Parse multiple commands without exiting the process

### ISSUE THREAD

**Issue by Bernd-L** [tj/commander.js#934]
# Goals

- Parse a final (immutable) string and take action accordingly
- Keep the process open (including on `-h` and `-v`, and even on incorrect input)
- Use commander.js as a library (will get called sometimes later, triggered by events)

## Edit:
Goals continuation:
- Get the output from `-v` and `-h` into a string instead of into the terminal

# Use case

A bot which should listen for new messages in a live-chat channel.
Upon the arrival of a new message, it will look for a specific prefix, then send the command to jommander.js to be parsed, and will take action accordingly.

Notice: The bot itself runs on a server in an automated manner and doesn't need nor expect any command-line arguments. The goal here is to parse several commands without the node process exiting because commander.js feels like it.

When a `prefix -v` comes along, it should respond. And it does - great!
But now my bot has made a `clean exit`.
Turns out commander.js decides when my app is done executing.

This behaviour is, of course, unacceptable, as the next user typing `prefix -v` into the chat will wait forever, for the bot is now shut down.

The fact that the bot needs to be reconnected to the chat-API and a database every time it starts doesn't make this situation easier.

# TL;DR

How does one parse multiple commands without having their app shut down in between?
How to get the output from `-v` and `-h` into a string instead of into the terminal?

---
PS: #443 Talks about something somewhat-related

**Comment by shadowspawn** [comment]
Thanks for the use case, and for the succinct `TL;DR` too. 

Related open issues about not calling exit: #443 #575 

And issues about allowing custom exit code, so similar but different use case: #188 #435

**Comment by shadowspawn** [comment]
I have started work on an exit override: https://github.com/tj/commander.js/issues/945#issuecomment-527412375

**Comment by shadowspawn** [comment]
There are improvements on the develop branch for the next release (#1040).

**Comment by shadowspawn** [comment]
v4.0.0-0 prerelease published: #1067

**Comment by shadowspawn** [comment]
v4.0.0 has been released.
